.github/workflows/debugger_exploration_testing.yml

```diff
@@ -0,0 +1,362 @@
+      PYTHONPATH: ../ddtrace/tests/debugging/exploration/
+    defaults:
+      run:
+        working-directory: starlette
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - uses: actions/checkout@v2
+        with:
+          path: ddtrace
+      - uses: actions/checkout@v2
+        with:
+          repository: encode/starlette
+          ref: 0.17.1
+          path: starlette
+      - name: Install ddtrace
+        run: pip install ../ddtrace
+      - name: Install dependencies
+        run: scripts/install
+      #Parameters for keyword expression skip 3 failing tests that are expected due to asserting on headers. The errors are because our context propagation headers are being added
+      #test_staticfiles_with_invalid_dir_permissions_returns_401 fails with and without ddtrace enabled
+      - name: Run tests
+        run: pytest -W ignore tests -k 'not test_request_headers and not test_subdomain_route and not test_websocket_headers and not test_staticfiles_with_invalid_dir_permissions_returns_401'
+
+  requests-testsuite-2_26_0:
+    name: requests 2.26.0
+    runs-on: "ubuntu-latest"
+    env:
+      DD_TESTING_RAISE: true
+    defaults:
+      run:
+        working-directory: requests
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+      - uses: actions/checkout@v2
+        with:
+          path: ddtrace
+      - uses: actions/checkout@v2
+        with:
+          repository: psf/requests
+          ref: v2.26.0
+          path: requests
+      - name: Install ddtrace
+        run: pip install ../ddtrace
+      - name: Install dependencies
+        run: "make init"
+      - name: MarkupSafe fix
+        run: pip install --upgrade MarkupSafe==2.0.1
+      - name: Run tests
+        run: PYTHONPATH=../ddtrace/tests/debugging/exploration/ pytest -p no:warnings tests
+
+  pylons-testsuite-1_0_3:
+    name: Pylons 1.0.3
+    runs-on: "ubuntu-latest"
+    env:
+      DD_TESTING_RAISE: true
+      PYTHONPATH: ../ddtrace/tests/debugging/exploration/
+    defaults:
+      run:
+        working-directory: pylons
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '2.7'
+      - uses: actions/checkout@v2
+        with:
+          path: ddtrace
+      - uses: actions/checkout@v2
+        with:
+          repository: pylons/pylons
+          ref: master
+          path: pylons
+      - name: Install ddtrace
+        run: pip install ../ddtrace
+      - name: Install test dependencies
+        run: pip install -e .[test]
+      - name: MarkupSafe fix
+        run: pip install --upgrade MarkupSafe==0.18 pip setuptools --force
+      - name: Disable failing tests
+        run: |
+          sed -i'' "s/test_detect_lang/detect_lang/g" tests/test_units/test_basic_app.py
+          sed -i'' "s/test_langs/langs/g" tests/test_units/test_basic_app.py
+      - name: Run tests
+        run: nosetests
```

Comment:
There's a lot of pulling the latest branch/release for each of the projects and their dependencies. This seems like it'll be a decent amount of work to maintain.

Do we wanna consider locking all these in?